### PR TITLE
fix: optional chaining on fallback fullstack date

### PR DESCRIPTION
**Pull request by naomi-lgbt**
Checklist:

- [X] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [X] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [X] My pull request targets the `main` branch of freeCodeCamp.
- [X] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

Fixes sentry error.

### CODE REVIEW

File: api-server/src/server/boot/certificate.js [hunk 1]
```diff
@@ -68,7 +68,7 @@ export function getFallbackFullStackDate(completedChallenges, completedDate) {
 
   const latestCertDate = completedChallenges
     .filter(chal => chalIds.includes(chal.id))
-    .sort((a, b) => b.completedDate - a.completedDate)[0].completedDate;
+    .sort((a, b) => b.completedDate - a.completedDate)[0]?.completedDate;
```
Comment (ShaunSHamilton) [inline api-server/src/server/boot/certificate.js:71]:
```suggestion
    .sort((a, b) => b.completedDate - a.completedDate)?.[0]?.completedDate;
```

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
Did you do this on purpose?

File: api-server/src/server/boot/certificate.js (hunk 1 above)
Comment (nhcarrigan) [reply api-server/src/server/boot/certificate.js:71]: `.filter` and `.sort` return empty arrays on no result, right? When would it be undefined?

Comment (ojeytonwilliams) [reply api-server/src/server/boot/certificate.js:71]: This is fine as is.

**Review by ojeytonwilliams — APPROVED** [review]

**Review by ShaunSHamilton — APPROVED** [review]

**Pull request merged**